Sources/StreamChat/Controllers/CurrentUserController/CurrentUserController.swift

```diff
@@ -94,17 +98,39 @@ public class _CurrentChatUserController<ExtraData: ExtraDataTypes>: Controller,
     init(client: _ChatClient<ExtraData>, environment: Environment = .init()) {
         self.client = client
         self.environment = environment
-        startObserving()
     }
     
-    private func startObserving() {
+    override public func synchronize(_ completion: ((_ error: Error?) -> Void)? = nil) {
+        startObservingIfNeeded()
+        
+        if case let .localDataFetchFailed(error) = state {
+            callback { completion?(error) }
+            return
+        }
+        
+        guard let currentUserId = currentUser?.id else {
+            completion?(ClientError.CurrentUserDoesNotExist())
+            return
+        }
+        
+        currentUserUpdater.loadUser(currentUserId) { error in
+            self.state = error == nil ? .remoteDataFetched : .remoteDataFetchFailed(ClientError(with: error))
+            self.callback { completion?(error) }
+        }
```

Comment:
We agreed we're not going to do this and make just the call for devices, since that the only thing from the user data that's not being updated via events.

Reply:
Discussed on slack, will change to updateDevices() :)